src/PIL/EpsImagePlugin.py

```diff
@@ -243,6 +245,34 @@ def check_required_header_comments():
                 msg = 'EPS header missing "%%BoundingBox" comment'
                 raise SyntaxError(msg)
 
+        def _read_comment(s):
+            nonlocal reading_trailer_comments
+            try:
+                m = split.match(s)
+            except re.error as e:
+                msg = "not an EPS file"
+                raise SyntaxError(msg) from e
+
+            if m:
+                k, v = m.group(1, 2)
+                self.info[k] = v
+                if k == "BoundingBox":
+                    if v == "(atend)":
+                        reading_trailer_comments = True
+                    elif not self._size or reading_trailer_comments:
```

Comment:
```suggestion
                    elif not self._size or (
                        trailer_reached and reading_trailer_comments
                    ):
```
I've realised I made a mistake in https://github.com/nopperl/Pillow/pull/1 - the last BoundingBox has priority not just if there was an "(atend)" value, but if there was an "(atend)" value AND we are now in the trailer.